File: TuringSuite/TuringSuite.Core/MachineTransitionSimple.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace TuringSuite.Core
{
    public class MachineTransitionSimple : IStateTransitionDescription<int, byte>
    {
        public byte FromSymbol { get; set; }

        public int FromState { get; set; }

        public byte WriteSymbol { get; set; }

        /// <summary>
        /// Gets or sets the next state. If the next state is to halt,
        /// this value should be negative.
        /// </summary>
        public int NextState { get; set; }

        public bool NextStateHalts { get; set; }

        public int MoveOffsetX { get; set; }

        public override string ToString()
        {
            return $"({FromState},{FromSymbol}) -> ({WriteSymbol},{MoveOffsetX},{NextState})";
        }
    }
}

File: TuringSuite/TuringSuite.Web/Static/js/MachineTransitionSimple.js
"use strict";

class MachineTransitionSimple {
	constructor(values) {
		this.FromSymbol = 0;
		this.FromState = 0;
		this.WriteSymbol = 0;
		this.NextState = 0;
		this.NextStateHalts = false;
		this.MoveOffsetX = 0;
		
		if (values) {
			if (values.hasOwnProperty("FromSymbol")) { this.FromSymbol = values.FromSymbol; }
			if (values.hasOwnProperty("FromState")) { this.FromState = values.FromState; }
			if (values.hasOwnProperty("WriteSymbol")) { this.WriteSymbol = values.WriteSymbol; }
			if (values.hasOwnProperty("NextState")) { this.NextState = values.NextState; }
			if (values.hasOwnProperty("NextStateHalts")) { this.NextStateHalts = values.NextStateHalts; }
			if (values.hasOwnProperty("MoveOffsetX")) { this.MoveOffsetX = values.MoveOffsetX; }
		}
	}
	
	toString() {
		return `(${this.FromState},${this.FromSymbol}) -> (${this.WriteSymbol},${this.MoveOffsetX},${this.NextState})`;
	}
}